python_modules/dagster/dagster/core/decorators.py

```diff
@@ -331,8 +331,23 @@ def transform(info, inputs):
             elif len(output_defs) == 1:
                 yield Result(value=result, output_name=output_defs[0].name)
             elif result is not None:
-                # XXX(freiksenet)
-                raise Exception('Output for a solid without an output.')
+                raise Exception(
```

Comment:
we should use DagsterInvariantViolationError for a user-facing error of this nature